File: epi_judge_java/epi/MinimumPointsCoveringIntervals.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.EpiUserType;
import epi.test_framework.GenericTest;

import java.util.Comparator;
import java.util.List;

public class MinimumPointsCoveringIntervals {

    @EpiTest(testDataFile = "minimum_points_covering_intervals.tsv")
    public static Integer findMinimumVisits(List<Interval> intervals) {
        if (intervals.isEmpty()) return 0;
        intervals.sort(Comparator.comparingInt(i -> i.right));
        int ans = 1;
        int latestCover = intervals.get(0).right;
        for (Interval interval : intervals) {
            if (interval.left > latestCover) {
                ans++;
                latestCover = interval.right;
            }
        }
        return ans;
    }

    @EpiUserType(ctorParams = {int.class, int.class})
    public static class Interval {
        public int left, right;

        public Interval(int l, int r) {
            this.left = l;
            this.right = r;
        }
    }

    public static void main(String[] args) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "MinimumPointsCoveringIntervals.java",
                                new Object() {
                                }.getClass().getEnclosingClass())
                        .ordinal());
    }
}

File: epi_judge_cpp/minimum_points_covering_intervals.cc
#include <vector>

#include "test_framework/generic_test.h"
#include "test_framework/serialization_traits.h"

using namespace std;

struct Interval {
    int left, right;
};

int FindMinimumVisits(vector<Interval> intervals) {
    if (intervals.empty()) return 0;
    sort(intervals.begin(), intervals.end(), [](auto &a, auto &b) {
        return a.right < b.right;
    });
    int ans = 1;
    int curr = intervals[0].right;
    for (const auto &[left, right] : intervals) {
        if (curr < left) {
            ans++;
            curr = right;
        }
    }
    return ans;
}

namespace test_framework {
    template<>
    struct SerializationTrait<Interval> : UserSerTrait<Interval, int, int> {
    };
} // namespace test_framework

int main(int argc, char *argv[]) {
    std::vector<std::string> args{argv + 1, argv + argc};
    std::vector<std::string> param_names{"intervals"};
    return GenericTestMain(args, "minimum_points_covering_intervals.cc",
                           "minimum_points_covering_intervals.tsv",
                           &FindMinimumVisits, DefaultComparator{}, param_names);
}
